src/btree/bt_sync.c

```diff
@@ -238,10 +238,12 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
         __wt_atomic_store_enum(&btree->syncing, WT_BTREE_SYNC_RUNNING);
 
         /*
-         * Reset the number of obsolete time window pages to let the eviction threads continue
-         * marking the clean obsolete time window pages as dirty once the checkpoint is finished.
+         * Reset the number of obsolete time window pages to let the eviction threads and checkpoint
+         * cleanup operation to continue marking the clean obsolete time window pages as dirty once
+         * the checkpoint is finished.
          */
-        btree->obsolete_tw_pages = 0;
+        btree->eviction_obsolete_tw_pages = 0;
```

Comment:
like the comment I add in this [pr](https://github.com/wiredtiger/wiredtiger/pull/10856). Will this variable be used in other threads? If so, we need to mark this as shared variable.

Reply:
Yes I agree, my PR contains the fix and will be merged soon. Then we can sync the code.

Reply:
I merged the develop and fixed the new code.